state/state.go

```diff
@@ -127,12 +127,57 @@ func (s *State) Copy() *State {
 	}
 }
 
+var (
+	errNilGenesisDoc = errors.New("no genesisDoc was found")
+
+	genesisDBKey = []byte("genesis-doc")
+)
+
+// GenesisDoc is the accessor to retrieve the genesisDoc associated
+// with a state. If the state has no set GenesisDoc, it fetches from
+// its database the JSON marshaled bytes keyed by "genesis-doc", and
+// parses the GenesisDoc from that memoizing it for later use.
+// If you'd like to change the value of the GenesisDoc, invoke SetGenesisDoc.
+func (s *State) GenesisDoc() (*types.GenesisDoc, error) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if s.cachedGenesisDoc == nil {
+		retrGenesisDocBytes := s.db.Get(genesisDBKey)
```

Comment:
what `retr` stands for? can't we just call it `bytes` or `jsonBytes`

Reply:
retr --> retrieve*
Sure, I'll update that to your suggestion, thanks.